src/tutorial/index.rst

```diff
@@ -435,30 +421,40 @@ Below you can see how the message output function may be assembled. Normally, ag
     }
     """
     ...
+
+  .. code-tab:: py Agent Python
+
+    ...
+    # Agent Function to output the agents ID and position in to a 2D spatial message list
+    @pyflamegpu.agent_function
+    def output_message(message_in: pyflamegpu.MessageNone, message_out: pyflamegpu.MessageSpatial2D):
+        message_out.setVariableUInt("id", pyflamegpu.getID())
+        message_out.setLocation( \
+            pyflamegpu.getVariableFloat("x"), \
+            pyflamegpu.getVariableFloat("y"))
+        return pyflamegpu.ALIVE
+    ...
     
 Next the message input agent function is implemented, two new concepts are introduced here: the message input iterator and accessing environment properties.
 
 Each FLAME GPU message type provides unique methods for accessing messages, in this case we are using the :class:`MessageSpatial2D<flamegpu::MessageSpatial2D>` type. Refer to the :ref:`agent communication guide<Device Agent Communication>` for details of other messaging format's usage.
 
-The only way to access spatial messaging types is via an iterator, which returns all messages in a Moore neighbourhood about the provided search location. This means, that all messages within the originally specified search radius will be returned, however it is necessary for the user to filter out the small number of messages which fall outside of this radius that will also be returned. Furthermore, agent's will also be returned their own message, so may wish to filter out that too.
+The only way to access spatial messaging types is via an iterator, which returns all messages in a Moore neighbourhood (discretised by the message radius) about the provided search location. This means, that all messages within the originally specified search radius will be returned, however it is necessary for the user to filter out messages which are contained within the Moor neighbour but fall outside of this radius that are also returned. Furthermore, agent's will also be returned their own message, so may wish to filter by outputting the originating agent id to the message.
 
-The spatial message iterator is accessed using :func:`FLAMEGPU->message_in()<flamegpu::MessageSpatial2D::In::operator()>`, this takes two ``float`` parameters specifying the search origin. Normally this will be passed directly to a C++ range-based for loop (e.g. ``or (const auto &message : FLAMEGPU->message_in(my_x, my_y)) {``), allowing the returned messages to be iterated.
+The spatial message iterator is accessed using :func:`FLAMEGPU->message_in()<flamegpu::MessageSpatial2D::In::operator()>` (or via the ``message_in`` agent function argument in Python), this takes two ``float`` parameters specifying the search origin. Normally this will be passed directly to a C++ range-based for loop, allowing the returned messages to be iterated.
```

Comment:
Pure python/agent python?

Reply:
agent python

Reply:
For consistency I have used *pure python* to describe the fact that the functions are written is a subset of *pure* python (as opposed to Python with C++ strings) but consistently named the subset of Python as *Agent Python*